### PR TITLE
Windows console fix

### PR DESCRIPTION
**Pull request by mribbons**
windows: disable extra console

1. Added additional code to win.cc to get tests to write to console
(These are probably in the wrong place, let me know if this can be improved)
3. modified cli to use SSC::Process when launching app, so std pipes can be captured and output
4. console can be reenabled using --w32-console

### CODE REVIEW

**Comment by mribbons** [comment]
Worth mentioning:
Linux CI still failing due to double free bug, https://github.com/socketsupply/socket/pull/172 should resolve
Windows CI still hangs. I have ran this code against later code and there is no hang, let's get this and #166 merged in then see how things are.

File: src/process/win.cc [hunk 1]
```diff
@@ -7,6 +7,27 @@
 
 namespace SSC {
 
+SSC::String FormatError(DWORD error, SSC::String source) {
```
Comment (heapwolf) [inline src/process/win.cc:10]: nice, much needed

**Review by heapwolf — APPROVED** [review]
this seems good to go, one thing to consider is, do we want to use the standard logging function and make it output nicely to windows console? right now it spits out garbage instead of colorized ansi sequences.

File: src/process/win.cc (hunk 1 above)
Comment (mribbons) [reply src/process/win.cc:10]: Agree, we want to move this to a more central location though, eventually.

**Comment by mribbons** [comment]
>  right now it spits out garbage instead of colorized ansi sequences.

It works now!! There was something wrong with the default terminal window, maybe it wasn't in the right mode.

File: src/app/app.cc [hunk 2]
```diff
@@ -87,8 +87,10 @@ namespace SSC {
     }
 #elif defined(_WIN32)
     if (isDebugEnabled()) {
-      fclose(console);
-      FreeConsole();
+      std::cout << "PostQuitMessage(0)" << std::endl;
```
Comment (jwerle) [inline src/app/app.cc:90]: do we need this here?

File: src/desktop/main.cc [hunk 3]
```diff
@@ -142,6 +142,13 @@ MAIN {
     // launched from the `ssc` cli
     app.fromSSC = s.find("--from-ssc") == 0 ? true : false;
 
+    if (!app.w32ShowConsole) {
+      app.w32ShowConsole = s.find("--w32-console") == 0 ? true : false;
+      if (app.w32ShowConsole) {
```
Comment (jwerle) [inline src/desktop/main.cc:147]: maybe this check should happen below? this `if` block?

File: src/window/win.cc [hunk 4]
```diff
@@ -993,6 +993,14 @@ namespace SSC {
                           return S_OK;
                         }
 
+                        if (msg.name == "stdout") {
```
Comment (jwerle) [inline src/window/win.cc:996]: why add this here instead of changing it in `bridge.cc` to handle windows?

File: src/desktop/main.cc (hunk 3 above)
Comment (mribbons) [reply src/desktop/main.cc]: Cleaned up code

File: src/window/win.cc (hunk 4 above)
Comment (mribbons) [reply src/window/win.cc]: Resolved, works via stdWrite now

**Review by jwerle — APPROVED** [review]

**Pull request merged**